### PR TITLE
Playlist Selector

### PR DESCRIPTION
**Pull request by featherbear**
## Context

[Corresponding Pivotal Tracker story](https://www.pivotaltracker.com/story/show/175369337)

## Description 💬

Playlist selection wrapper for `MasterPlaylist` and `Playlist` components.  
Allows for the selection of a playlist from Spotify

Contains commits from `genre_splitting`

### Usage

```html
<Selector [text?=string] [playlist?=Playlist] component=Component />
```

* `text` - Prompt override
* `playlist` - Playlist override
* `component` - The component you want to wrap

## Checklist ✅:

- [x] I assigned the story to myself on Pivotal Tracker before I started working on this
- [x] Added a link to the user story context
- [x] Added a description of my changes
- [x] Assigned at least two reviewers

### CODE REVIEW

**Review by CarlinWilliamson — CHANGES_REQUESTED** [review]
Could you include private playlists and hide the sub-playlists until a master playlist is selected

**Comment by z5160345** [comment]
I'm seeing 5 checks failed and the preview not compiling

**Comment by featherbear** [comment]
> I'm seeing 5 checks failed and the preview not compiling

Yep, currently working on the features.
Will action for a re-review when ready

**Comment by featherbear** [comment]
@CarlinWilliamson  @z5160345  changes have been made, please review

**Review by CarlinWilliamson — APPROVED** [review]
10/10

**Comment by z5160345** [comment]
Is removing the popup on hover of a track intended?

**Comment by featherbear** [comment]
> 
> 
> Is removing the popup on hover of a track intended?

Which changeset are you looking at? I don't I touched that.
It should eventually be introduced in a feature branch. This current PR addresses only the playlist selection, and changes to the framework

File: src/components/TrackEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      setTrack(await props.track.expand())
+    })()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  return (
+    <TableRow /* key={track?.id || id++} */>
+      <TableCell component='th' scope='row'>
+        {track ? (
+          /*<Track track={track} />*/ track.name
```
Comment (featherbear) [inline src/components/TrackEntry.tsx:24]: `Track` component not developed yet. See [story #175033867](https://www.pivotaltracker.com/story/show/175033867)

Comment (z5160345) [reply src/components/TrackEntry.tsx:24]: ah ok cool

**Review by z5160345 — APPROVED** [review]
wow thats a lot of code

**Pull request merged**